From 11f940a8671617da823aabd93fc4077c3a6a31b7 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 18 Sep 2024 15:55:39 +0300
Subject: [PATCH] [FSW-9220] fs1: boot_desc: use spi offsets array

Since the SPI offset of the primary image and the secondary image are
different, but the pri and sec manifests are identical, both offsets
are stored in each manifest.

When arcboot implementation will mature, the stable image index will
be stored in the bootinfo/coldboot regs, and ppboot will use it
when reading the spi offset from the descriptor.

Change-Id: I3c882a260c04b09c6f3a66cbdaa1deb55ac2a2cf
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 5443b93..8ac5ee5 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -512,7 +512,8 @@
 u32 dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
 			  struct boot_desc *desc)
 {
-	desc->spi_offset = desc_ptr->offset;
+	/* TODO: use stable index when bootinfo implementation is final in arcboot */
+	desc->spi_offset = desc_ptr->spi_offset[0];
 	desc->img_size = desc_ptr->size;
 	desc->loadaddr = ((u64)desc_ptr->load_high << 32) | desc_ptr->load_low;
 	desc->desc_ver = desc_ptr->desc_ver;
